logstash-core/lib/logstash/settings.rb

```diff
@@ -200,25 +225,61 @@ def coerce(value)
           raise ArgumentError.new("could not coerce #{value} into a boolean")
         end
       end
+    end
 
-      def set(value)
-        coerced_value = coerce(value)
-        validate(coerced_value)
-        @value = coerce(coerced_value)
-        @value_is_set = true
-        @value
+    class Numeric < Coercible
+      def initialize(name, default=nil, strict=true)
+        super(name, ::Numeric, default, strict)
+      end
+
+      def coerce(v)
+        return v if v.is_a?(::Numeric)
```

Comment:
not much more than that really, end of the day for this settings code is more scoped, so the "risk" or surprises is lower. If we encounter this as a problem we can alware rethink.